src/software/new_geom/angle.h

```diff
@@ -168,6 +168,13 @@ class Angle final
      */
     constexpr Angle clamp() const;
 
+    /**
+     * Verifies if this angle is kept between [−π, π]
+     *
+     * @return if the angle is kept between [−π, π] for invariant
+     */
+    bool invariantHolds(const Angle &angle) const;
```

Comment:
Two quick things here:
1) Why do we need to maintain this invariant?
2) If we elect to keep this, could we name it something like `angleBetweenPiAndNegativePi`? It's a bit more verbose, but I think `invariantHolds` is a bit generic.

Reply:
1. The original ticket's 2nd acceptance criteria stated: ` Add the invariant to Angle that it is always kept between [−π, π] and change the functions accordingly`. The invariantHolds in the Angle class is there to satisfy this acceptance criteria.
2. I can change it if the consensus is to keep the invariant.

Also, CI seems to be failing and I'm unsure why ATM, will look later.

Reply:
summoning @jsn0716

Reply:
> Also, CI seems to be failing and I'm unsure why ATM, will look later.

Are you seeing `bazel test //software/...` fail on your local machine?

Reply:
I suppose this was never discussed further, initial discussion here: https://github.com/UBC-Thunderbots/Software/pull/986#discussion_r338876274

The benefit would be that we wouldn't need to worry about clamping after each operation like adding and subtracting.

Reply:
Hmm, I think that it makes more sense to enforce clamping where we need it, rather then just all the time. Given how often this class is used, any extra work in it will probably have a notably overall performance hit.

Reply:
Since most trig functions implemented internally shouldn't care about angles being in `[-pi, pi]`, I think this invariant would only really matter when the user calls `toRadians` or `toDegrees` to get the value of the angle. What if we only clamped the angle value to `[-pi, pi]` when one of those functions is called?

Reply:
Pinging to see if any conclusion was drawn and if we want to keep the `invariantHolds`/`angleBetweenPiAndNegativePi` function since it's the only blocker on the PR.

Reply:
Apologies for the delay on this. I think the best option here would be to simply have a `clampBetweenPiAndNegativePi` function that we call at the start of any function that requires that assumption.

Reply:
And I think in order to address any performance concerns, I think the only functions that would need that clamping would be the `toRadians` and `toDegrees` functions. Unless @jsn0716 you had other ideas when you made the ticket.

Reply:
Nope, that sounds good to me

Reply:
To clarify @garethellis0, do you mean calling `clamp` or `angleBetweenPiAndNegativePi`? ATM, it sounds like you just want to call `clamp` in the beginning of both `toRadians` and `toDegrees`.

Reply:
Hmm, I don't think we should modify the actual angle unless we need to. Why would we need to clamp in `toRadians` or `toDegrees`? I'm still not sure I see where the need for it is? The only case I can think of where we might clamp is when printing maybe, but even then I would not modify the actual angle because IMO printing should never mutate state.

Reply:
Agree re: printing. If not in `toRadians` or `toDegrees`, then is there anywhere at all we'd want to call this automatically? Otherwise it doesn't sound like there's any invariant we (@garethellis0 ) want to maintain because of performance penalties.

Reply:
Correct. As it stands, I think printing is the only use-case, unless @jsn0716 knows of others in geometry code?

Reply:
Sorry just to clarify, if there were no performance penalties, do you think the invariant is worth maintaining?

Reply:
Nope, don't see any real benefit.

Reply:
Ok can we get some consensus / a vote here for if people think the invariant is worth it? @jonathanlew @akhilveeraghanta @jsn0716 @kelvinkoon @scveloso

Reply:
IMO if it's not used anywhere else ATM, then it can be omitted. 
If it becomes more relevant in the future, then it shouldn't be difficult to add the invariant in.